cluster-api-gcp/cloud/google/machineactuator.go

```diff
@@ -522,17 +544,16 @@ func (gce *GCEClient) instanceIfExists(machine *clusterv1.Machine) (*compute.Ins
 }
 
 func (gce *GCEClient) providerconfig(providerConfig string) (*gceconfig.GCEProviderConfig, error) {
-	obj, gvk, err := gce.codecFactory.UniversalDecoder().Decode([]byte(providerConfig), nil, nil)
+	var obj gceconfigv1.GCEProviderConfig
+	_, _, err := gce.serializer.Decode(
+		[]byte(providerConfig),
+		&schema.GroupVersionKind{Group:"gceproviderconfig", Version:"v1alpha1", Kind:"GCEProviderConfig"},
```

Comment:
Can you reference to the values for these where gceproviderconfig is defined instead of re-hardcoding group version and kind here?